src/subdomains/staking/application/services/liquidity-management.service.ts

```diff
@@ -186,6 +186,14 @@ export class LiquidityManagementService {
   }
 
   // --- HELPER METHODS --- //
+  private async getPendingWithdrawals(): Promise<Withdrawal[]> {
+    const stakingList = await this.withdrawalService.getStakingWithPendingWithdrawals();
+    return stakingList.reduce(
+      (prev, curr) => prev.concat(curr.getPendingWithdrawals().map((w) => Object.assign(w, { staking: curr }))),
+      [],
+    );
+  }
```

Comment:
intended revert of your changes?

Reply:
Yes, I'm not really happy with it ... but we need it for further improvements. Maybe we can discuss it in a quiet hour 😉